### ISSUE TITLE
[Semantic tokens] Provide sufficient contrast across components

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#7798]
### Violation: 
 Ensure parts of graphical objects essential for understanding content have sufficient contrast 

![image](https://github.com/Esri/calcite-design-system/assets/5023024/cb06553a-0ee5-4c6f-9e29-6d6ea39cbc2a)

### WCAG Reference: 
- [1.4.11 Non-test Contrast](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) 

### Severity: 
 5 

### Media Type: 
 Color and Contrast 

--- 
### Areas for Remediation: 
- Title: Rating - Ensure parts of graphical objects essential for understanding content have sufficient contrast - (2036570022) 
- Module: Rating 
**Issue**
There are graphical objects with a color contrast ratio below 3.00:1.

Foreground: #EDD317
Background: #F8F8F8 
Contrast ratio: 1.4:1

Examples include:
- Test 2. Average  Rating
- Test 3. Average with chip

**User Impact**
Users with low vision will have difficulty identifying this content.

**Code Reference** 
 ```
<svg class="svg" fill="currentColor" height="100%" viewBox="0 0 24 24" width="100%" xmlns="http://www.w3.org/2000/svg">
...
</svg>


 ``` 
**Suggestion**
Ensure parts of graphical objects essential for understanding content have sufficient contrast. The required minimum contrast ratio is 3.00:1.

Common examples of qualifying objects include lines in a chart, meaningful icons, and annotations within an image.

Graphics that require particular presentation to preserve their meaning are exempt from this requirement.

To evaluate color contrast, see the Accessible Color Picker extension: https://www.accessibility.dev/

## Calcite design solution
Update the following semantic tokens to resolve contrast issue:

**LIGHT MODE**
status/warning/default → `color/high-saturation/orange-yellow/h-oy-060`
status/warning/hover → `color/vibrant/orange-yellow/v-oy-180`
status/warning/press → `color/high-saturation/orange-yellow/h-oy-080`
**DARK MODE**
status/warning/default → `color/high-saturation/orange-yellow/h-oy-060`
status/warning/hover → `color/vibrant/orange-yellow/v-oy-120`
status/warning/press → `color/vibrant/orange-yellow/v-oy-140`

🔗 [Link to Figma specs](https://www.figma.com/design/rkpMoRvtpYb9SD9dYHXg2i/-Rating--star-color-accessibility?node-id=546-4774&t=8on7xjVMIR0xgmov-11)

![Image](https://github.com/user-attachments/assets/9d1800b6-132d-4f61-a52c-903ec327dc7f)

 
--- 
### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022. 

**monday.com sync:** #6619738221

**Comment by geospatialem** [comment]
For design consideration, should we accommodate color contrast with `rating`, or does the outline of each item provide sufficient contrast and we can close the above? cc @ashetland @SkyeSeitz

**Comment by SkyeSeitz** [comment]
Thanks, @geospatialem. Would like to explore a potential change to the yellow star color displayed for the `average` prop.

**Comment by jacqueskeet** [comment]
During the exploration of this, we realised that the `Read-only` property of this component is missing. We should explore adding it back as part of this ticket. Here's a design example of what it should look like. 

<img width="515" alt="Rating - Read only" src="https://github.com/Esri/calcite-design-system/assets/117104798/06e80dda-175e-4b25-9da5-e6972ec14f8b">

**Comment by jacqueskeet** [comment]
Following the design exploration, we decided to update the global warning color used in the Rating component. The Warning token color has been adjusted, providing sufficient contrast with the Rating component. 

![image](https://github.com/user-attachments/assets/34fd0b18-dec8-465a-86a4-d26d6a8f7055)

We're aware that this is a global change hence we identified two potential risks associated with this change:
	1. other components that uses this color;
	2. a potential contrast issue next to the danger color;

We looked at the components that are using the Warning color and found that there’s no immediate issues. 

![image](https://github.com/user-attachments/assets/2a5acce6-c835-45c6-99ff-e47268097125)

We also ran a few visual impairment contrast tests between the same components using the new warning color and the danger color and found that there’s no issues as well. 
 
 Example of Protanopia
![image](https://github.com/user-attachments/assets/83dda470-f814-4fd3-b484-1a093d0b7164)

**Comment by SkyeSeitz** [comment]
> During the exploration of this, we realized that the `Read-only` property of this component is missing. We should explore adding it back as part of this ticket...


As mentioned above, in addition to updating the warning colors, can we also implement an updated `read-only` styling to improve distinction? This change would use the `star-f` icon with a `border.1` fill to distinguish itself from the outlined `star` used in the default state. 
<img width="489" alt="image" src="https://github.com/user-attachments/assets/9a89d5bd-fce3-467a-8607-54b48b801103">

**Comment by jcfranco** [comment]
@geospatialem @SkyeSeitz @jacqueskeet Can we create a separate issue for the design improvements (e.g., read-only support, global warning color tweak)? This was originally scoped to the contrast of the existing design. Alternatively, we could repurpose this issue to focus on the improvements and add the a11y requirements to the acceptance criteria if that’s preferable.

**Comment by geospatialem** [comment]
Skye created https://github.com/Esri/calcite-design-system/issues/11837 to accommodate the component's read only styling. 💪🏻

**Comment by DitwanP** [comment]
🍠 Verified
